src/main/scala/nl.knaw.dans.easy.sword2/FilesPermission.scala

```diff
@@ -43,8 +44,9 @@ object FilesPermission {
         case cce: ClassCastException => log.error("No file permission elements in set", cce)
         case ioe: IOException => log.error(s"Could not set file permissions on $path", ioe)
         case se: SecurityException => log.error(s"Not enough privileges to set file permissions on $path", se)
+        case iae: IllegalArgumentException => log.error(s"Incorrect permissions input string: $permissions, could not set permission on $path", iae)
         case e => log.error(s"Unexpected exception on $path", e)
-      } getOrElse FileVisitResult.TERMINATE
+      } unsafeGetOrThrow
```

Comment:
Though not part of this issue, I'd rather see something with Vistor in the class name, it doesn't do anything recursively. @rvanheest an opinion about something like `FilePermissionVistor`?

Reply:
im also pretty sure i cannot just throw here

Reply:
@jo-pol what about `RecursiveFilePermissionVisitor`?

Reply:
i can live with that name